core/string/src/lib.rs

```diff
@@ -871,13 +926,23 @@ impl Clone for JsString {
     #[inline]
     fn clone(&self) -> Self {
         if let UnwrappedTagged::Ptr(inner) = self.ptr.unwrap() {
-            // SAFETY: The reference count of `JsString` guarantees that `raw` is always valid.
+            // SAFETY: `NonNull` and the constructions of `JsString` guarantees that `raw` is always valid.
```

Comment:
You also need to explain why it is safe to access the union as a `read_only`; a `Cell` is readable as an `usize` as long as we don't try to mutate the pointed variable.